### PR TITLE
Add Traces view to Service Details

### PR DESCRIPTION
**Pull request by jotak**
Fixes https://github.com/kiali/kiali/issues/3071

- Make class AppTracesC more generic, allowing app/service/workload input (and rename it: TracesComponent)
- TracesFetcher now fetches different endpoints depending on target kind
- "JaegerItemC" now receives a "focusElement" instead of app (and is renamed: TraceDetails)
- Remove now unused cleanServiceSelector function
- Define server endpoints API for fetching service/workload based traces
- Add Traces tab to ServiceDetailsPage

Backend PR: https://github.com/kiali/kiali/pull/3169

### CODE REVIEW

**Comment by jotak** [comment]
WIP: server-side pending work

File: src/components/JaegerIntegration/TracesComponent.tsx [hunk 1]
```diff
@@ -82,6 +85,9 @@ class AppTracesC extends React.Component<AppTracesProps, AppTracesState> {
       'none';
 
     const traceId = HistoryManager.getParam(URLParam.JAEGER_TRACE_ID) || undefined;
+    if (this.props.targetKind === 'app') {
+      this.focusApp = this.props.target;
```
Comment (jotak) [inline src/components/JaegerIntegration/TracesComponent.tsx:89]: note for self: that should go under componentDidUpdate

**Review by hhovsepy — APPROVED** [review]
Verified:
Screenshot from 2020-09-03 14-58-17

File: src/config/Config.ts [hunk 2]
```diff
@@ -127,7 +127,9 @@ const conf = {
         `api/iter8/namespaces/${namespace}/experiments/${name}`,
       istioPermissions: 'api/istio/permissions',
       jaeger: 'api/jaeger',
-      jaegerTraces: (namespace: string, app: string) => `api/namespaces/${namespace}/apps/${app}/traces`,
+      appTraces: (namespace: string, app: string) => `api/namespaces/${namespace}/apps/${app}/traces`,
+      serviceTraces: (namespace: string, svc: string) => `api/namespaces/${namespace}/services/${svc}/traces`,
+      workloadTraces: (namespace: string, wkd: string) => `api/namespaces/${namespace}/workloads/${wkd}/traces`,
```
Comment (lucasponce) [inline src/config/Config.ts:132]: I think this is not yet in the backend, isn't it ?

File: src/services/Api.ts [hunk 3]
```diff
@@ -328,8 +328,16 @@ export const getJaegerInfo = () => {
...
-  return newRequest<JaegerResponse>(HTTP_VERBS.GET, urls.jaegerTraces(namespace, app), params, {});
+export const getAppTraces = (namespace: string, app: string, params: TracingQuery) => {
+  return newRequest<JaegerResponse>(HTTP_VERBS.GET, urls.appTraces(namespace, app), params, {});
+};
+
+export const getServiceTraces = (namespace: string, service: string, params: TracingQuery) => {
+  return newRequest<JaegerResponse>(HTTP_VERBS.GET, urls.serviceTraces(namespace, service), params, {});
+};
+
+export const getWorkloadTraces = (namespace: string, workload: string, params: TracingQuery) => {
```
Comment (lucasponce) [inline src/services/Api.ts:339]: I think this is left here, this endpoint is not yet in the backend.

**Review by lucasponce — CHANGES_REQUESTED** [review]
Frontend code looks ok.
I have a couple of comments in the backend code, but I've tested locally and it seems ok.

File: src/config/Config.ts (hunk 2 above)
Comment (jotak) [reply src/config/Config.ts:132]: yes I'm a bit ahead of time for this one ... first i thought i would do both tasks at the same time..
I can remove if you prefer, but anyway, I'll start to work on the workload case right after that one

**Review by lucasponce — APPROVED** [review]

**Pull request merged**